### PR TITLE
[TRTLLM-11268][feat] Video temporal compression to Nemotron Nano and RADIO

### PR DESCRIPTION
**Pull request by 2ez4bz**
## Summary by CodeRabbit

* **New Features**
  * Added video preprocessing with aspect-ratio preservation for improved frame handling.
  * Introduced temporal compression support for efficient video frame grouping and processing.
  * Extended vision models to support video inputs with configurable temporal and spatial parameters.

* **Tests**
  * Added comprehensive test coverage for video preprocessing utilities and temporal compression workflows.

## Description

Implement tubelet-based temporal compression for video inputs, matching the Megatron-LM / vLLM video processing pipeline. T consecutive frames are grouped into tubelets before embedding, reducing the token count by a factor of `video_temporal_patch_size`.

Key additions:
- Aspect-ratio-preserving video frame resize and normalization
- Separate video embedder in RADIO ViT for tubelet projection
- Tubelet-aware token counting, frame separators, and EVS paths
- Fix align_corners=True -> False in position embedding interpolation

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

To see a list of available CI bot commands, please comment `/bot help`.

### CODE REVIEW

File: tensorrt_llm/_torch/models/modeling_nemotron_nano.py [hunk 1]
```diff
@@ -607,6 +802,36 @@ def __init__(
...
+        if target_num_patches is not None:
+            self.video_target_num_patches = target_num_patches
+        elif target_img_size is not None:
+            base_patches = math.ceil(target_img_size / self.patch_size)
+            self.video_target_num_patches = base_patches * base_patches
+        else:
+            self.video_target_num_patches = None
+
+        # Normalization mean/std for video frames (used by video_to_pixel_values).
+        self._video_norm_mean = torch.tensor(getattr(config, "norm_mean", [0.5, 0.5, 0.5])).reshape(
```
[the remark this answers is not in the document]
Comment (2ez4bz) [reply tensorrt_llm/_torch/models/modeling_nemotron_nano.py]: You're right. Let me just raise.

**Review by Wanli-Jiang — APPROVED** [review]

File: tensorrt_llm/_torch/models/modeling_radio.py [hunk 2]
```diff
@@ -288,7 +353,7 @@ def window_select(pos_embed):
             max_dim = max(input_dims)
             pos_embed = F.interpolate(pos_embed.float(),
                                       size=(max_dim, max_dim),
-                                      align_corners=True,
```
Comment (yechank-nvidia) [inline tensorrt_llm/_torch/models/modeling_radio.py]: Question) Is this change for video support? Or were we computing wrongly so far?

File: tensorrt_llm/_torch/models/modeling_nemotron_nano.py [hunk 3]
```diff
@@ -765,34 +1043,35 @@ def get_num_tokens_per_video(
...
-                self.get_num_tokens_per_image(
-                    image=frame,
-                    video_pruning_rate=None,
-                    max_num_tiles=VIDEO_MAX_NUM_TILES,
-                    **kwargs,
-                )
-                for frame in video
-            )
+            # No pruning: tokens_per_unit * num_tubelets + special tokens.
+            num_special = len(self.get_mm_special_token_ids())
```
[the remark this answers is not in the document]
Comment (yechank-nvidia) [reply tensorrt_llm/_torch/models/modeling_nemotron_nano.py]: Feel free to resolve this, when you fix this one, please.

**Review by yechank-nvidia — APPROVED** [review]
